engine/src/test/java/org/hibernate/validator/test/constraints/ConstraintValidatorContextImplTest.java

```diff
@@ -51,7 +57,9 @@ public void testIterableKeyed() {
 				.addConstraintViolation();
 
 		List<ConstraintViolationCreationContext> constraintViolationCreationContextList = context.getConstraintViolationCreationContexts();
-		assertMessageAndPath( constraintViolationCreationContextList.get( 0 ), message, "foo[test]" );
+		assertMessageAndPath( constraintViolationCreationContextList.get( 0 ), message, pathWith()
+				.property( "foo" )
+				.property( null, true, "test", null ) );
```

Comment:
Instead of the gazillion of parameters, should we rather have dedicated methods such as `listProperty()` etc., each taking the required parameters only. Definitely a separate thing, though.